runtime-sdk/modules/evm/src/backend.rs

```diff
@@ -174,6 +174,35 @@ impl<'ctx, C: Context, Cfg: Config> EVMBackend for Backend<'ctx, C, Cfg> {
     }
 }
 
+pub(crate) trait EVMBackendExt {
+    fn random_bytes(&self, num_words: u64, pers: &[u8]) -> Vec<u8>;
+}
+
+impl<T: EVMBackendExt> EVMBackendExt for &T {
+    fn random_bytes(&self, num_words: u64, pers: &[u8]) -> Vec<u8> {
+        (*self).random_bytes(num_words, pers)
+    }
+}
+
+impl<'ctx, C: Context, Cfg: Config> EVMBackendExt for Backend<'ctx, C, Cfg> {
+    fn random_bytes(&self, num_words: u64, pers: &[u8]) -> Vec<u8> {
+        // Refuse to generate more than 2 KiB in one go.
+        // EVM memory gas is checked only before and after calls, so we won't
+        // see the quadratic memory cost until after this call uses its time.
+        let num_words = num_words.min(64);
```

Comment:
I think that we'll see people hit this upper limit relatively frequently, as EVM gas costs disincentivize using sub-word sized values, and 52 is the size of a deck of cards. Maybe we should set this to 32 and force people to be efficient? (@kostko)